### ISSUE TITLE
Can't create cluster with not registered master

### ISSUE THREAD

**Issue by piosz** [kubernetes/kubernetes#36167]
**Kubernetes version** (use `kubectl version`): HEAD

**Environment**: GCE

**What happened**:
When I set up `REGISTER_MASTER_KUBELET=false` I can see the following message:
```
... calling validate-cluster
Waiting for 3 ready nodes. 1 ready nodes, 1 registered. Retrying.
Found 4 nodes, but expected 3. Your cluster may not behave correctly.
Found 4 node(s).
NAME                           STATUS                     AGE
kubernetes-master              Ready,SchedulingDisabled   19s
kubernetes-minion-group-jifk   NotReady                   16s
kubernetes-minion-group-n5ao   NotReady                   15s
kubernetes-minion-group-y419   NotReady                   17s
```

**What you expected to happen**:
master should not be registered

**Comment by piosz** [comment]
cc @kubernetes/sig-cluster-lifecycle @mikedanese @roberthbailey

**Comment by roberthbailey** [comment]
Looks like the env var isn't being respected any longer by the master scripts (but the logic for # of expected nodes did adjust correctly).

**Comment by roberthbailey** [comment]
/cc @jszczepkowski 

I think this may have broken in #33655 because it added

```
+  if [[ -z "${KUBERNETES_MASTER_NAME:-}" ]]; then
 +    KUBERNETES_MASTER_NAME="${MASTER_NAME}"
 +  fi
```

which unconditionally sets the field that tells a kubelet to register with the master outside of the guard that checks `REGISTER_MASTER_KUBELET`.

**Comment by jszczepkowski** [comment]
Fix in progress: #36214.

There was a problem caused by unconditional setting `KUBELET_APISERVER` on the master in #33655.
